static/src/javascripts/projects/common/modules/discussion/upvote.js

```diff
@@ -0,0 +1,76 @@
+define([
+    'common/utils/fastdom-promise',
+    'common/utils/report-error'
+], function (
+    fastdom,
+    reportError
+) {
+    var RECOMMENDATION_CLASS = 'js-recommend-comment';
+    var TOOLTIP_CLASS = 'js-rec-tooltip';
+    var HIDE_TOOLTIP_CLASS = 'tooltip-box-hidden';
+
+    function handle (target, container, user, discussionApi) {
+        if (!user) {
+            return showSignInTooltip(target);
+        } else if (isOpenForRecommendations(container)) {
+            var id = target.getAttribute('data-comment-id');
+
+            return Promise.all([
+                setClicked(target),
+                discussionApi.recommendComment(id)
+            ])
+            .then(function () {
+                return setRecommended(target);
+            })
+            .catch(function (ex) {
+                unsetClicked(target);
+                reportError(ex, {
+                    feature: 'comments-recommend'
+                });
+            });
+        }
+    }
+
+    function isOpenForRecommendations (element) {
+        return !!element.querySelector('.d-discussion--recommendations-open');
```

Comment:
You could use `getElementsByClassName` which is much faster, but I don't know if there is a polyfill for IE8